src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorManagedRuleOverrideObject.cs

```diff
@@ -0,0 +1,58 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.FrontDoor.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Commands.FrontDoor.Models;
+
+namespace Microsoft.Azure.Commands.FrontDoor.Cmdlets
+{
+    /// <summary>
+    /// Defines the New-AzFrontDoorManagedRuleOverrideObject cmdlet.
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "FrontDoorManagedRuleOverrideObject"), OutputType(typeof(PSAzureManagedRuleOverride))]
+    public class NewAzureRmFrontDoorManagedRuleOverrideObject : AzureFrontDoorCmdletBase
+    {
+        /// <summary>
+        /// Rule ID
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "Rule ID")]
+        public string RuleId { get; set; }
+
+        /// <summary>
+        /// Override Action
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Override Action")]
+        public PSAction Action { get; set; }
```

Comment:
Use a string with a completer, this will future-proof the API from future breaking changes and still allow the user to select from appropriate values.

Reply:
Why would you prefer a string other than an Enum ? If I add anything to the Enum, then the next version will pick it up. If I was deleting anything from the Enum, that would be a breaking change regardless.

Reply:
This takes some of the urgency from releasing and installing a new version every time the enum values are updated - users would be able to use new values without you havign to release a new version fo the cmdlets (or them havign to install a new version).

Reply:
Not really, because I still need to create an object of type PSAzureManagedRuleOverride (which is then used in other objects). Then the model gets converted (and eventually a string is actually sent).

(line 348 of ModelExtensions.cs)
Action = ruleOverride.Action.HasValue ? ruleOverride.Action.Value.ToString() : null,

So, I understand what you are proposing but it would requiring touching pretty much every file and I still believe the user would have to install a new version if we ever add a value (since the API version is still hardcoded). If I want this to be a string, then the model has to be a string, then the model extension has to change, then I also have to change the parameters in all other cmdlets that have been using Enums (to be consistent). At the end, if we ever add a value, the user might have to install a new package version anyway (because of the hard coded version).

Also, this particular Enum (PSAction) is likely not going to change for a while (if ever) and any change would still require a new swagger with a new API version.

Reply:
Is this modeled as a string in yoru API, or as an enum?  If this is modeled as a string in your SDK, then you should change this to string and update the types to match.

If it is modeled as an enum in the SDK, then thsi is OK for now, but in the next swagger iteration, this should be modeled as an expandable enum (modelAsString: true)

Reply:
It's an Enum in the model right now.